File: Algorithm/Backjun/bronze5/b2884/b2884.c
// 알람 시계
// https://www.acmicpc.net/problem/2884

#include <stdio.h>

int main(void) {
  int h, m;
  scanf("%d %d", &h, &m);
  if (m >= 45) {
    m = m-45;
  }
  else {
    m = 60-(45-m);
    if (h == 0) {
      h = 23;
    }
    else h = h-1;
  }

  printf("%d %d", h, m);

  return 0;
}

File: Algorithm/Backjun/bronze5/b2884/b2884.py
# 알람 시계
# https://www.acmicpc.net/problem/2884

h, m = map(int, input().split())

if (m < 45):
    h -= 1;
    m += 15;
    if h < 0: h = 23
else: m -= 45

print(h, m)
